analyzers/tests/SonarAnalyzer.UnitTest/MetadataReferences/NuGetMetadataFactory.Package.cs

```diff
@@ -20,152 +20,135 @@
 
 using System;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.IO;
 using System.Linq;
 using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Packaging;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
 
 namespace SonarAnalyzer.UnitTest.MetadataReferences
 {
     public static partial class NuGetMetadataFactory
     {
         private class Package
         {
-            public string Id { get; }
-            public string Version { get; }
-            public string Runtime { get; }
+            private readonly string id;
+            private readonly string runtime;
+            private string version;
 
             public Package(string id, string version, string runtime)
             {
-                Id = id;
-                Version = version;
-                Runtime = runtime;
+                this.id = id;
+                this.version = version;
+                this.runtime = runtime;
             }
 
             public string PackageDirectory()
             {
-                var runtimePath = Runtime == null ? string.Empty : $"runtimes\\{Runtime}\\";
-                var versionFolder = Version == Constants.NuGetLatestVersion
-                        ? SortedPackageFolders().Select(x => Path.GetFileName(x).Substring(Id.Length + 1)).Last(x => char.IsNumber(x[0]))
-                        : Version;
-                return Path.Combine(PackagesFolder, $@"{Id}.{versionFolder}\{runtimePath}lib");
+                var runtimePath = runtime == null ? string.Empty : $"runtimes\\{runtime}\\";
+                var combinedPath = Path.Combine(PackagesFolder, id, $"Sonar.{version}", runtimePath);
+                return Path.GetFullPath(combinedPath);
             }
 
             public void EnsurePackageIsInstalled()
             {
-                if (Version == Constants.NuGetLatestVersion)
+                var latestVersionObtained = version == Constants.NuGetLatestVersion;
+                if (latestVersionObtained)
                 {
-                    if (IsCheckForLatestPackageRequired())
-                    {
-                        Install();
-                        WriteNextCheckTime();
-                    }
-                    else
-                    {
-                        LogMessage($"Skipping check for latest NuGet");
-                    }
+                    version = GetLatestVersion().Result;
```

Comment:
Let's do this in the constructor and make it readonly. It's a weird side effect.